spec/features/collection_templates/template_index_spec.rb

```diff
@@ -0,0 +1,34 @@
+# The tests in this file verify that the index page contains the expected elements
+
+describe 'When visiting the template index page', js:true do
+  before do
```

Comment:
I think you should wrap all these contents in another context block: `context 'when there is one record' do...`